File: benchmark/comparison/results/adv_ts/06.ts
function running_sum(lst: number[]): number[] {
  const result: number[] = [];
  let sum = 0;
  for (const n of lst) {
    sum += n;
    result.push(sum);
  }
  return result;
}

function format_list(lst: number[]): string {
  return `[${lst.join(", ")}]`;
}

for (const test of [[1, 2, 3, 4], [], [5], [-1, 2, 3]]) {
  console.log(format_list(running_sum(test)));
}

File: benchmark/comparison/results/adv_py/06.py
def running_sum(lst):
    result = []
    total = 0
    for x in lst:
        total += x
        result.append(total)
    return result

def fmt(lst):
    return "[" + ", ".join(str(x) for x in lst) + "]"

for test in [[1, 2, 3, 4], [], [5], [-1, 2, 3]]:
    print(fmt(running_sum(test)))
